frontend/src/components/editor/renderers/vertical-layout/vertical-layout.tsx

```diff
@@ -19,24 +22,36 @@ const VerticalLayoutRenderer: React.FC<VerticalLayoutProps> = ({
   mode,
 }) => {
   const { invisible } = useDelayVisibility(cells.length, mode);
+  const [showCode, setShowCode] = useState(false);
+  // Show code if there is at least one cell with code
+  const canShowCode = mode === "read" && cells.some((cell) => cell.code);
   return (
-    <VerticalLayoutWrapper
-      className="sm:pt-8"
-      invisible={invisible}
-      appConfig={appConfig}
-    >
+    <VerticalLayoutWrapper invisible={invisible} appConfig={appConfig}>
       {cells.map((cell) => (
         <VerticalCell
           key={cell.id}
           cellId={cell.id}
           output={cell.output}
           status={cell.status}
+          code={cell.code}
           stopped={cell.stopped}
+          showCode={showCode && canShowCode}
           errored={cell.errored}
           mode={mode}
           interrupted={cell.interrupted}
         />
       ))}
+      {canShowCode && (
+        <div className="fixed m-4 left-0 bottom-0">
+          <Button
+            onClick={() => setShowCode((prev) => !prev)}
+            data-testid="show-code"
+          >
+            <Code2Icon className="w-4 h-4 mr-2" />
+            {showCode ? "Hide code" : "Show code"}
+          </Button>
+        </div>
+      )}
```

Comment:
This is great! Can you put it in the blue banner instead of here, to the left of the "run or edit" button?